doc/dev/debugging.rst

```diff
@@ -403,4 +403,44 @@ The recompilation starts after the given checkpoint stage.
 
     replace_ir(f, "HLOLoweringPass", new_ir)
     f(2.0)
-    >> 8.0
\ No newline at end of file
+
+C Executable Generation
+=================
+
+Catalyst provides a way to generate a c executable that calls the shared object of the given
+qjit-decorated function.
+Let's use cross references :)
+```suggestion
+The function :func:`~.debug.compile_executable` takes a qjit-decorated function and
+concrete arguments to that function as input.
+It returns the path to the output executable file.
+Users can also use `debug.print_memref` to add information to stdout.
+
+The following example is a square function.
+Here we are using debug.print_memref to print the information of the result from ``y``.
+The executable will be saved in the directory for intermediate results if ``keep_intermediate=True``.
+Otherwise, the executable will appear in the Catalyst project root
+
+.. code-block:: python
+    @qjit
+    def f(x):
+        y = x*x
+        debug.print_memref(y)
+        return y
+    f(5)
+>>> MemRef: base@ = 0x64fc9dd5ffc0 rank = 0 offset = 0 sizes = [] strides = [] data =
+... 25
+... 25
```

Comment:
This formatting is showing up correctly, I think there needs to be an empty line between the code block and console part (and the >>> should be on the command not the output):
```suggestion
.. code-block:: python

    @qjit
    def f(x):
        y = x*x
        debug.print_memref(y)
        return y

>>> f(5)
MemRef: base@ = 0x64fc9dd5ffc0 rank = 0 offset = 0 sizes = [] strides = [] data =
25
```